server/game/cards/locations/05/oldwyk.js

```diff
@@ -7,30 +7,42 @@ class OldWyk extends DrawCard {
         this.reaction({
             when: {
                 onChallenge: (event, challenge) =>
-                    challenge.attackingPlayer === this.controller
-                    && challenge.challengeType === 'power'
+                    challenge.attackingPlayer === this.controller &&
+                    challenge.challengeType === 'power' &&
+                    this.anyDrownedGodInDeadPile()
             },
             cost: ability.costs.kneelSelf(),
             handler: () => {
-                var card = _.last(this.controller.deadPile.filter(c => c.hasTrait('Drowned God')));
-
-                if(!card) {
-                    this.game.addMessage('{0} finds no character in their dead pile to put into play with {1}', this.controller, this);
-                    return;
-                }
+                let card = _.last(this.controller.deadPile.filter(c => c.hasTrait('Drowned God')));
 
                 this.controller.putIntoPlay(card);
                 this.game.currentChallenge.addAttacker(card);
 
-                this.game.addMessage('{0} uses {1} to put into play {2} as an attacker from their dead pile', this.controller, this, card);
+                this.game.addMessage('{0} kneels {1} to put {2} into play from their dead pile as an attacker', 
+                                      this.controller, this, card);
 
-                this.untilEndOfChallenge(ability => ({
-                    match: card,
-                    effect: ability.effects.returnToHandOrDeckBottom()
-                }));
+                this.game.once('afterChallenge:interrupt', (event, challenge) => this.resolveAfterChallenge(challenge, card));
```

Comment:
Hmm. That seems in contradiction to the FAQ for Old Wyk: http://thronesdb.com/card/05028
> The "If you win the challenge..." effect resolves before any Interrupts or Reactions to winning/losing the challenge can be played.

If it happens between Interrupts and Reactions as you stated, then Old Wyk was correct before (the effect should be reprocessed just before Reactions). If the FAQ is correct, it's a question of whether it should happen before all Interrupts (including forced and cancellation interrupts) or just before player triggered interrupts.